nextjs/packages/next/server/render.tsx

```diff
@@ -381,6 +382,14 @@ function checkRedirectValues(
   }
 }
 
+function normalizeRedirectValues(redirect: Redirect) {
+  const dest = redirect.destination
+  if (dest && typeof dest === 'object') {
+    const resolvedDest = resolveHrefOnServer(dest)
```

Comment:
We can re-use `formatWithValidation` function here — https://github.com/blitz-js/blitz/blob/canary/nextjs/packages/next/shared/lib/utils.ts#L567
```suggestion
    const resolvedDest = formatWithValidation(dest)
```

Reply:
actually, using formatWithValidation was my first idea. it works, but with a catch! i'll capture a video later to demonstrate you

Reply:
with formatWithValidation:
https://user-images.githubusercontent.com/41614937/142560305-21131556-e348-4430-8fc0-1ab1c76967f6.mp4

with resolveHrefOnServer(i don't really like the name though):
https://user-images.githubusercontent.com/41614937/142560313-589eafcf-1977-4d0e-89a4-c1859b59eebb.mp4

Reply:
Okay, got it. Thanks for the recordings! 

@flybayer It looks like we actually need to use `resolveHref` function as we're probably interested in the `interpolatedAs` value: https://github.com/blitz-js/blitz/blob/canary/nextjs/packages/next/shared/lib/router/router.ts#L336

Reply:
@s-r-x ok makes sense. Is there any issue with hardcoding `{ pathname: '/', asPath: '/' }`? Is there any edge cases this could cause a problem for? It would be good to add a clear comment explaining this for future folks.

Reply:
@flybayer, @beerose  i replaced that hacky router mock with real server router(basically i just remove that resolveHrefOnServer util and revert the type of resolveHref router from { pathname: string, asPath: string } back to NextRouter). final result is basically the same, but the solution is less hacky. and i think there is no need for additional comments anymore.